File: patterns/javascript-patterns/pattern83.js
// Pattern 83: ...

//     *
//    - -
//   -   -
//  -     -
// -       -
// -       -
//  -     -
//   -   -
//    - -
//     *

const n = 10;

let px = Math.floor(n / 2);
let py = Math.floor(n / 2);

for (let i = 1; i <= n; i++) {
    let row = '';

    for (let j = 1; j <= n - 1; j++) {
        if ((i === 1 || i === n) && j === n / 2) {
            row += '*';
        } else if (j === px || j === py) {
            row += '-';
        } else {
            row += ' ';
        }
    }

    if (i === n / 2) {
        let t = px;
        px = py;
        py = t;
    } else {
        px--;
        py++;
    }

    console.log(row);
}

File: patterns/python-patterns/pattern83.py
# Pattern 83: ...

#     *
#    - -
#   -   -
#  -     -
# -       -
# -       -
#  -     -
#   -   -
#    - -
#     *

n = 10
px = n // 2
py = n // 2

for i in range(1, n + 1):
    for j in range(1, n):
        if (i == 1 or i == n) and j == n // 2:
            print("*", end="")
        elif j == px or j == py:
            print("-", end="")
        else:
            print(" ", end="")
    if i == n // 2:
        t = px
        px = py
        py = t
    else:
        px -= 1
        py += 1
    print()
